src/xtopcom/xcommon/src/xtoken.cpp

```diff
@@ -37,10 +39,10 @@ xtop_token & xtop_token::operator=(xtop_token && other) {
 xtop_token::xtop_token(common::xsymbol_t symbol) : symbol_{ std::move(symbol) } {
 }
 
-xtop_token::xtop_token(std::uint64_t const amount, common::xsymbol_t symbol) : amount_{ amount }, symbol_ { std::move(symbol) } {
+xtop_token::xtop_token(evm_common::u256 const amount, common::xsymbol_t symbol) : amount_{ amount }, symbol_ { std::move(symbol) } {
```

Comment:
Can we use evm_common::u256 for TOP? The namespace is a bit confusing.

Reply:
> Can we use evm_common::u256 for TOP? The namespace is a bit confusing.

yes. but at the storage level, it depends on which token value is saved. for TOP, u64 is enough.